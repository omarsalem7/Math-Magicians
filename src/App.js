import './App.css';
import React from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import Header from './components/Header';
import Calculator from './components/Calculator';
import Home from './components/Home';
import Quote from './components/Quote';

const App = () => (
  <Router>
    <Header />
    <h1>omar</h1>
    <div className="App">
      <Routes>
        <Route exact path="/" element={<Home />} />
        <Route exact path="/calculator" element={<Calculator />} />
        <Route exact path="/quote" element={<Quote />} />
      </Routes>
    </div>
  </Router>
);
export default App;
