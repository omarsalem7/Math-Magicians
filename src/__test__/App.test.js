import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import App from '../App';

test('check if app contain this test', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Welcome to our page/i);
  expect(linkElement).toBeInTheDocument();
});

describe('Testing App component', () => {
  it('App renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
