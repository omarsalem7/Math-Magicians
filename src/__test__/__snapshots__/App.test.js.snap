// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component App renders correctly 1`] = `
Array [
  <ul
    className="header"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "orange",
            "margin": "1rem",
            "textDecoration": "none",
          }
        }
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
        style={
          Object {
            "color": "orange",
            "margin": "1rem",
            "textDecoration": "none",
          }
        }
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="/quote"
        onClick={[Function]}
        style={
          Object {
            "color": "orange",
            "margin": "1rem",
            "textDecoration": "none",
          }
        }
      >
        Quote
      </a>
    </li>
  </ul>,
  <h2
    className="title"
  >
    Math Magicians
  </h2>,
  <h1>
    omar
  </h1>,
  <div
    className="App"
  >
    <div
      className="home"
    >
      <h2>
        Welcome to our page
      </h2>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Bibendum neque egestas congue quisque egestas diam in arcu cursus. Non nisi est sit amet facilisis magna. Amet tellus cras adipiscing enim eu turpis egestas. Nullam vehicula ipsum a arcu cursus vitae congue mauris. Et malesuada fames ac turpis egestas sed. Elit ullamcorper dignissim cras tincidunt lobortis
      </p>
    </div>
  </div>,
]
`;
